aks/locals.tf

```diff
@@ -1,23 +1,123 @@
+resource "random_string" "name" {
+  length  = 3
+  special = false
+  numeric = false
+  upper   = false
+}
+
 locals {
+  roxiepool = {
+    ultra_ssd         = false
+    node_os           = "ubuntu"
+    node_type         = "gp"
+    node_type_version = "v2"
+    node_size         = var.aks_node_sizes.roxie
+    single_group      = false
+    min_capacity      = 1
+    max_capacity      = 3
+    labels = {
+      "lnrs.io/tier" = "standard"
+      "workload"     = "roxiepool"
+    }
+    taints = []
+    tags   = {}
+  }
+
+  node_groups0 = {
+    thorpool = {
+      ultra_ssd         = false
+      node_os           = "ubuntu"
+      node_type         = "gp"      # gp, gpd, mem, memd, stor
+      node_type_version = "v2"      # v1, v2
+      node_size         = var.aks_node_sizes.thor
+      single_group      = false
+      min_capacity      = 3
+      max_capacity      = 6
```

Comment:
TLH: Added new variable, 'aks_capacity' which defines the minimum and maximum number of nodes in each node pool.

Reply:
TLH: This one will take a while to figure out because of the many different node/VM sizes. The following link lists all the node/VM sizes possibly: https://github.com/hpccsystems-solutions-lab/tlh-oss-terraform-azurerm-aks/blob/make-logging-and-monitoring-optional/modules/node-groups/modules/node-group/local.tf

Reply:
Agreed.  I suggest you make an Executive Decision and support only a few (3?) different sizes.  You can fit requirements into VM sizes the best way you can.

Note that the user does not need to know about your decision.

Reply:
TLH: I will use that approach. Thanks.

Reply:
Fixed. Now thorpool capacity is set based on the user's values for 'thor_num_workers', 'thor_max_jobs', 'thor_worker_cpus', and thorpool node size selected.

Reply:
Continuation of my last reply. Also, the above mentioned calculations also sets the number of cpus and ram each thor worker should have.